torch/distributed/optim/named_optimizer.py

```diff
@@ -299,15 +299,19 @@ def init_state(self) -> None:
         self.step(closure=None)
 
     def _pre_load_state_dict(self, state_dict) -> Dict[str, Any]:
+        # TODO(chienchin): This API should be FSDP agnostic and should support
+        # general user hooks.
```

Comment:
Do you mean we need a contract here which let user to register a hook `optim_state_dict_to_load` and check whether there is a hook or not? Or how do we want to make it support general user hooks?